net/wireguard/src/opnsense/scripts/OPNsense/Wireguard/genkey.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+# Copyright (c) 2018 Michael Muenz <m.muenz@gmail.com>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+TMPDIR="/tmp"
+GENPRIV="/usr/local/bin/wg genkey"
+GENPUB="/usr/local/bin/wg pubkey"
+
+cleanup() {
+  # Delete old files
+  rm -f $TMPDIR/wireguard.*
+}
+
+private() {
+	# Generate a private key and put it to /tmp
+	umask 077 && ${GENPRIV} > ${TMPDIR}/wireguard.priv && cat ${TMPDIR}/wireguard.priv
+}
+
+public() {
+	# Generate a public key and put it to /tmp
+	cat ${TMPDIR}/wireguard.priv | ${GENPUB} > ${TMPDIR}/wireguard.pub && cat ${TMPDIR}/wireguard.pub
+}
+
+case "$1" in
+	private)
+                cleanup
+		private
+		;;
+	public)
+		public
+		;;
+esac
```

Comment:
If you do that in PHP with pipes, you don't have to care so much about file system security.